main.go

```diff
@@ -355,3 +343,25 @@ func run() int {
 	}
 
 }
+
+func getTimeout(r *http.Request, module config.Module, offset float64) (timeoutSeconds float64, err error) {
+	// If a timeout is configured via the Prometheus header, add it to the request.
+	if v := r.Header.Get("X-Prometheus-Scrape-Timeout-Seconds"); v != "" {
+		var err error
+		timeoutSeconds, err = strconv.ParseFloat(v, 64)
+		if err != nil {
+			return 0, err
+		}
+	}
+	if timeoutSeconds == 0 {
+		timeoutSeconds = 10
+	}
+
+	if module.Timeout.Seconds() < timeoutSeconds && module.Timeout.Seconds() > 0 {
```

Comment:
Shouldn't we do the subtraction before this?

Reply:
Not sure.

As per the linked issue, the idea is that `offset` has no effect in case the user explicitly set per-module timeout. So, we only do subtraction if there's no explicitly set timeout. The requirement of the specified timeout not exceeding Prometheus scrape-timeout is also met (again, without taking offset into account then).

Reply:
We should still take the offset into account when comparing against the scrape timeout, otherwise we're not allowing for network time to the exporter.

Reply:
Would `if module.Timeout.Seconds() < (timeoutSeconds - offset) && module.Timeout.Seconds() > 0 {` be OK?

Reply:
Yes, but it'd be clearer to do the subtraction before the if.

Reply:
Done 👌